File: src/Stronghold.Application/Features/Users/UserResponse.cs
namespace Stronghold.Application.Features.Users;

public class UserResponse
{
    public int Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? ProfileImageUrl { get; set; }
    public string Role { get; set; } = string.Empty;
    public int Level { get; set; }
    public int XP { get; set; }
    public int TotalGymMinutes { get; set; }
}

File: stronghold_desktop/lib/features/users/models/user_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_response.freezed.dart';
part 'user_response.g.dart';

@freezed
abstract class UserResponse with _$UserResponse {
  const factory UserResponse({
    required int id,
    required String firstName,
    required String lastName,
    required String username,
    required String email,
    String? phone,
    String? address,
    String? profileImageUrl,
    required String role,
    required int level,
    required int xp,
    required int totalGymMinutes,
  }) = _UserResponse;

  factory UserResponse.fromJson(Map<String, dynamic> json) =>
      _$UserResponseFromJson(json);
}

@freezed
abstract class PagedUserResponse with _$PagedUserResponse {
  const factory PagedUserResponse({
    required List<UserResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedUserResponse;

  factory PagedUserResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedUserResponseFromJson(json);
}
